### PR TITLE
Aditi Mohan - ADB Tenders - 11th December 2021

### PR DESCRIPTION
**Pull request by Aditi-Mohan**
Implemented Scraping, Standardisation, and Cleaning.

Some results generated by the scraper are exported into the file - sample_results.json

Run spider using command -  scrapy crawl abd_spider 
or to store results - scrapy crawl abd_spider -o <filename>

### CODE REVIEW

File: dummy-data-product/src/dependencies/scraping/abd/abd/spiders/abd_spider.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+from scrapy import Request
+import logging
+from datetime import datetime as dt
+
+
+
+class AbdSpiderSpider(scrapy.Spider):
+    name = 'abd_spider'
+    allowed_domains = ['www.adb.org']
+    start_urls = ['https://www.adb.org/projects?page=0/']
```
Comment (KayvanShah1) [inline dummy-data-product/src/dependencies/scraping/abd/abd/spiders/abd_spider.py:70]: You have scraped data from an incorrect page. 
You are supposed to scrape tenders data from https://www.adb.org/projects/tenders

**Review by KayvanShah1 — COMMENTED** [review]
### **Feedback:**
- Overall the class is well written keeping modularity in mind. 
- Though it seems unclear where data scraped from the previous run is loaded, updated, and saved. 
- Albeit you have worked on a different page, the final output data is in a suitable form and well structured.

### Next steps:
- Implementation of data processing as a separate process
- Devising an efficient data management and storage strategy
- Maintaining data from previous runs and intermediary steps for backup and recovery.

**Comment by KayvanShah1** [comment]
Hi Aditi,
According to our technical discussion yesterday, we will go through your improvised solution on 11th Dec 2021 and will schedule a meeting to discuss the same. Once you are done resolving the above comments kindly ping in the discussion below to notify us for a review.

File: dummy-data-product/src/dependencies/scraping/abd/abd/spiders/abd_spider.py (hunk 1 above)
Comment (KayvanShah1) [reply dummy-data-product/src/dependencies/scraping/abd/abd/spiders/abd_spider.py:70]: For your reference pasting the link to wiki documentation below...

https://github.com/Taiyo-ai/pt-mesh-pipeline/wiki

You can find more information about the scripting practices and class structures here.

**Comment by Aditi-Mohan** [comment]
I have added the features discussed last time for the final submission.

**Comment by gnommer** [comment]
Saw the new commits. I have some comments

• Can we implement s3 and elastic search using plugins to simplify the data pipeline. Cause seems more effective to publish the data using plugins
• Docker image might need to be built to support running scrapy pipeline.
• We should profile the resources possible and see if it’s possible to incorporate a state to rerun, resume and scrape new entries. It think it’s possible in scrapy just need to configure it.
• In future we could use splash if we wanna scrape JS based rendered data.

This is all that I can say @KayvanShah1 any other comments from your side ?

**Pull request closed without merging**